src/imex/transfer.rs

```diff
@@ -132,8 +134,12 @@ impl BackupProvider {
                 // Explicit drop to move the guards into this future
                 drop(paused_guard);
                 drop(dbfile);
-                res
+                res.map_err(|err| format!("{err:#}"))
             })
+            .map_err(|err| format!("{err}"))
+            .and_then(futures::future::ready)
```

Comment:
why is this needed?

Reply:
The version before this PR did this flattening of the error in the `Future::poll` implementation by using `?` there.  This is the equivalent but does it on the future itself: it flattens the task's JoinError into the error of the future the task is running.  It just makes it a bit easier to use the future output.